pkg/credentialprovider/gcp/metadata.go

```diff
@@ -151,10 +163,16 @@ func (g *dockerConfigUrlKeyProvider) Provide() credentialprovider.DockerConfig {
 // Enabled implements a special metadata-based check, which verifies the
 // storage scope is available on the GCE VM.
 func (g *containerRegistryProvider) Enabled() bool {
-	value, err := credentialprovider.ReadUrl(metadataScopes+"?alt=json", g.Client, metadataHeader)
-	if err != nil {
+	if !onGCEVM() {
 		return false
 	}
+	// Given that we are on GCE, we should keep retrying until the metadata server responds.
+	for {
+		value, err := credentialprovider.ReadUrl(metadataScopes+"?alt=json", g.Client, metadataHeader)
```

Comment:
Yeah. Sorry I did not test the patch. Wanted to post it to share it with you. I have cleaned it up now. PTAL